### ISSUE TITLE
build fail with 

### ISSUE THREAD

**Issue by TomLeeQuanta** [rhboot/efivar#172]
t2s@quanta:~/efivar/docs$ cd ..
t2s@quanta:~/efivar$ make
make[1]: Entering directory '/home/t2s/efivar/src'
make[2]: Entering directory '/home/t2s/efivar/src'
make[2]: Nothing to be done for 'deps'.
make[2]: Leaving directory '/home/t2s/efivar/src'
gcc -Og -flto -g3 -Wall -Wextra  -Werror  -std=gnu11 -funsigned-char -fvisibility=hidden -specs=/home/t2s/efivar/src/include/gcc.specs -fno-merge-constants  -fPIC  -DLIBEFIVAR_VERSION=37 -D_GNU_SOURCE -I/home/t2s/efivar/src/include/ -c -o lib.o lib.c
In file included from efivar.h:15,
                 from lib.c:17:
util.h:45:28: error: expected identifier or ‘(’ before ‘__asm__’
   45 | # define VERSION(sym, ver) __asm__(".symver " # sym "," # ver)
      |                            ^~~~~~~
lib.c:32:1: note: in expansion of macro ‘VERSION’
   32 | VERSION(_efi_set_variable, _efi_set_variable@libefivar.so.0)
      | ^~~~~~~
util.h:45:28: error: expected identifier or ‘(’ before ‘__asm__’
   45 | # define VERSION(sym, ver) __asm__(".symver " # sym "," # ver)
      |                            ^~~~~~~
lib.c:49:1: note: in expansion of macro ‘VERSION’
   49 | VERSION(_efi_set_variable_variadic, efi_set_variable@libefivar.so.0)
      | ^~~~~~~
util.h:45:28: error: expected identifier or ‘(’ before ‘__asm__’
   45 | # define VERSION(sym, ver) __asm__(".symver " # sym "," # ver)
      |                            ^~~~~~~
lib.c:66:1: note: in expansion of macro ‘VERSION’
   66 | VERSION(_efi_set_variable_mode,efi_set_variable@@LIBEFIVAR_0.24)
      | ^~~~~~~
lib.c:85:1: error: ‘efi_set_variable’ aliased to undefined symbol ‘_efi_set_variable_mode’
   85 | efi_set_variable(efi_guid_t guid, const char *name, uint8_t *data,
      | ^~~~~~~~~~~~~~~~
make[1]: *** [/home/t2s/efivar/src/include/rules.mk:48: lib.o] Error 1
make[1]: Leaving directory '/home/t2s/efivar/src'
make: *** [Makefile:14: all] Error 2

Does anyone how to solve?

**Comment by nvzoll** [comment]
In lib.c, move all occurencies of macros `VERSION` like this:

Before:
```
int NONNULL(2, 3) PUBLIC
VERSION(_efi_set_variable_variadic, efi_set_variable@libefivar.so.0)
_efi_set_variable_variadic(efi_guid_t guid, const char *name, uint8_t *data,
			   size_t data_size, uint32_t attributes, ...)
{
    // ...
}
```

After:
```
int NONNULL(2, 3) PUBLIC
_efi_set_variable_variadic(efi_guid_t guid, const char *name, uint8_t *data,
			   size_t data_size, uint32_t attributes, ...)
{
    // ...
}
VERSION(_efi_set_variable_variadic, efi_set_variable@libefivar.so.0);
```

The issue was probably introduced by this commit: https://github.com/rhboot/efivar/commit/8a573944789c38ca08195538e5943e1f67e0de89

**Comment by TomLeeQuanta** [comment]
Thank you, it can build successfully
After I make install, I execute efivar

t2s@quanta:~/efivar$ efivar
efivar: /lib/x86_64-linux-gnu/libefivar.so.1: version `LIBEFIVAR_1.38' not found (required by efivar)

I found LIBEFIVAR_1.37 no 1.38
How to fix it?

**Comment by nvzoll** [comment]
It tries to load the library from the build host, not from the repository you built. For a quick fix, use bash command `export LD_LIBRARY_PATH=$(pwd):$LD_LIBRARY_PATH` inside the `src/` folder (where "libefivar.so" is located).

**Comment by TomLeeQuanta** [comment]
It can work, thank

**Comment by frozencemetery** [comment]
Probably fixed in c08bb8948f3f56f78b19d9f708cfeb3fcd3cc74d
